content/blog/2022-01-15-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md

```diff
@@ -0,0 +1,280 @@
+    sudo ubuntu-drivers autoinstall
+
+    sudo curl -fsSL https://get.docker.com -o get-docker.sh && sudo sh get-docker.sh &&
+    sudo usermod -aG docker ubuntu
+    sudo setfacl --modify user:ubuntu:rw /var/run/docker.sock
+    curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | sudo apt-key add -
+    curl -s -L https://nvidia.github.io/nvidia-docker/ubuntu20.04/nvidia-docker.list | sudo tee /etc/apt/sources.list.d/nvidia-docker.list
+    sudo apt update && sudo apt install -y nvidia-docker2
+    sudo systemctl restart docker
+    rm get-docker.sh
+
+    nvidia-smi
+    docker run --rm --gpus all -v "$PWD:/tpi" iterativeai/cml:0-dvc2-base1-gpu \
+        /bin/bash -c "cd /tpi; pip install -r requirements.txt; python src/train.py"
+    END
+}
+```
+
+Here, to run my training code I am passing my project's directory as a volume to
+a Docker image provided by Iterative (`iterativeai/cml:0-dvc2-base1-gpu`) which
+takes care of CUDA.
+
+## Ready… Set… Apply!
+
+Whether you want to go with the basic example, or the GPU-enabled training, you
+can run `terraform plan` to review what steps Terraform is going to take to make
+your desired infrastructure materialize. Don't worry, nothing is actually done
+at this point, but it's a good way to check for potential issues in the
+configuration.
+
+If there are no issues, run `terraform apply` to execute the actions proposed in
+the Terraform plan. You'll need to type `yes` to confirm that you indeed would
+like the plan to be executed.
+
+At this point you can get a cup of your favourite beverage, and let the
+Iterative Provider work its magic together with Terraform. They will allocate a
+remote machine for you, set up Docker, pull required images, install missing
+drivers, and finally run your `train.py`. Once the script finishes, the machine
+will be terminated.
+
+You can monitor what's going on by running
+`terraform refresh && terraform show`. Once you see that the task has
+successfully finished, go ahead and run `terraform destroy` to destroy all
+remote objects managed by your configuration and sync back your shared files. If
+you write metrics to a file, then you'll get the new metrics synced back to your
+local machine.
+
+Now if you want to try another experiment, you can change your code, run
+`terraform apply` again, and when the training is finished, commit your code
+together with the updated metrics.json. This can help you move from prototyping
+locally to leveraging more powerful cloud machines without the hassle of full
+MLOps setup. At the same time, once you're ready to start working on your
+production pipelines and CI/CD, this should also make the transition smoother.
+
+In this tutorial we have covered the simplest example with no GPU or Docker, and
+one that involves leveraging GPUs with the help of `nvidia-docker` and
+`iterativeai/cml:0-dvc2-base1-gpu`. In many cases, the next step in
```

Comment:
```suggestion
In this tutorial we have covered the simplest example without and with GPUs.
In many cases, the next step in
```

Reply:
Resolved in b0443d1e629d77d7b44cf9b3c9f444f80629913a